recipe/meta.yaml

```diff
@@ -19,45 +19,54 @@ outputs:
     script: build_base.bat  # [win]
     script: build_base.sh  # [not win]
     build:
-      skip: true  # [py<37]
+      skip: true  # [py<38]
       ignore_run_exports:
         - tk  # [linux]
+        - zlib
     requirements:
       build:
+        - python
+        - cross-python_{{ target_platform }}       # [build_platform != target_platform]
+        - numpy
         - pkg-config  # [not win]
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
       host:
         - python
-        - pip
+        - pip >=9.0.1
         - certifi >=2020.06.20
-        - freetype >=2.3
+        # freetype.pc requires zlib.pc
+        - freetype
+        - zlib
         # an explanation to the utilization of numpy can be found
         # here: https://github.com/AnacondaRecipes/matplotlib-feedstock/pull/13#issuecomment-1296704563
         - numpy   1.19  # [not (osx and arm64) and (py==37 or py==38 or py==39)]
         - numpy   1.21  # [(osx and arm64) and (py==38 or py==39)]
         - numpy   1.21  # [py==310]
         - setuptools
-        - setuptools_scm >=4
+        - setuptools_scm >=7
         - setuptools_scm_git_archive
         - tk  # [linux]
```

Comment:
Are they still needed? I do not see them in `setup.py`: https://github.com/matplotlib/matplotlib/blob/v3.6.2/setup.py#L303-L307

Reply:
I've found some documentation regarding `tk` and it is an optional dependency for `matplotlib`. See [here](https://matplotlib.org/stable/devel/dependencies.html#:~:text=Tk%20(%3E%3D%208.4%2C%20!%3D%208.6.0%20or%208.6.1)%3A%20for%20the%20Tk%2Dbased%20backends.%20Tk%20is%20part%20of%20most%20standard%20Python%20installations%2C%20but%20it%27s%20not%20part%20of%20Python%20itself%20and%20thus%20may%20not%20be%20present%20in%20rare%20cases).
`Setuptools_scm_git_archive` seems to play an essential role in our successful builds. I've noticed that when removed from the host section, all our builds began to fail.

Reply:
@marcoesters Here is a great comment about `tk` as a default backend https://github.com/conda-forge/matplotlib-feedstock/pull/157#issue-325602391